client/checkout/blocks/upe-fields.js

```diff
@@ -0,0 +1,167 @@
+
+const WCPayUPEFields = ( {
+	api,
+	activePaymentMethod,
+	stripe,
+	elements,
+	billing: { billingData },
+	eventRegistration: {
+		onPaymentProcessing,
+		onCheckoutAfterProcessingWithSuccess,
+	},
+	emitResponse,
+} ) => {
+	const [ paymentIntentId, setPaymentIntentId ] = useState( null );
+	const [ clientSecret, setClientSecret ] = useState( null );
+	const [ hasRequestedIntent, setHasRequestedIntent ] = useState( false );
+	const [ isUPEComplete, setIsUPEComplete ] = useState( false );
+
+	const testMode = getConfig( 'testMode' );
+	const testCopy = (
+		<p>
+			<strong>Test mode:</strong> use the test VISA card 4242424242424242
+			with any expiry date and CVC.
+		</p>
+	);
+
+	useEffect( () => {
+		if ( paymentIntentId || hasRequestedIntent ) {
+			return;
+		}
+
+		async function createIntent() {
+			const response = await api.createIntent();
+			setPaymentIntentId( response.id );
+			setClientSecret( response.client_secret );
+		}
+		setHasRequestedIntent( true );
+
+		createIntent();
+	}, [ paymentIntentId, hasRequestedIntent, api ] );
+
+	// When it's time to process the payment, generate a Stripe payment method object.
+	useEffect(
+		() =>
+			onPaymentProcessing( () => {
+				if ( PAYMENT_METHOD_NAME_CARD !== activePaymentMethod ) {
+					return;
+				}
+
+				if ( ! isUPEComplete ) {
+					return {
+						type: 'error',
+						message: 'Your payment information is incomplete.',
+					};
+				}
+			} ),
+		// not sure if we need to disable this, but kept it as-is to ensure nothing breaks. Please consider passing all the deps.
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+		[ activePaymentMethod, isUPEComplete ]
+	);
+
+	// Once the server has completed payment processing, confirm the intent of necessary.
+	useEffect(
+		() =>
+			onCheckoutAfterProcessingWithSuccess(
+				( { orderId, processingResponse: { paymentDetails } } ) => {
+					async function updateIntent() {
+						await api.updateIntent(
+							paymentIntentId,
+							orderId,
+							false
+						);
+
+						const paymentElement = elements.getElement(
+							PaymentElement
+						);
+
+						return confirmUPEPayment(
+							api,
+							paymentDetails.redirect_url,
+							paymentElement,
+							billingData,
+							emitResponse
+						);
+					}
+
+					updateIntent();
+				}
+			),
+		// not sure if we need to disable this, but kept it as-is to ensure nothing breaks. Please consider passing all the deps.
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+		[ elements, stripe, api, paymentIntentId ]
+	);
+
+	// Checks whether there are errors within a field, and saves them for later reporting.
+	const upeOnChange = ( event ) => {
+		setIsUPEComplete( event.complete );
+	};
+
+	const elementOptions = {
+		clientSecret,
+		classes: {
+			base: 'wcpay-card-mounted',
+		},
+		fields: {
+			billingDetails: {
+				address: {
+					postalCode: 'never',
+					country: 'never',
+				},
+			},
+		},
```

Comment:
Is there any reason why you only pass these specific fields, as the fields that the UPE expects we will collect ourselves? You are passing in the rest of the form fields [over here](https://github.com/Automattic/woocommerce-payments/blob/ff190b89037cac25b2ef488f743b9a08254c6fb1/client/checkout/blocks/confirm-upe-payment.js#L25-L42); for example, I think if you pass the `name` field as well, then Giropay will hide this field at checkout.

Reply:
Nope, just hadn't gone back to it after the initial set up. These should all be removed from the fields now with the latest push.